docs/user_v2/tutorials/using-existing-devices.rst

```diff
@@ -0,0 +1,164 @@
+below and place it in an ``epics_demo.py`` file:
+
+.. literalinclude:: ../examples/epics_demo.py
+    :language: python
+
+The top section of the file is explained in the Bluesky tutorial, but the bottom
+section is Ophyd specific.
+
+First of all we start up a specific EPICS IOC for the demo devices. This is only
+used in this tutorial:
+
+.. literalinclude:: ../examples/epics_demo.py
+    :language: python
+    :start-after: # Start IOC
+    :end-before: # Create v1
+
+Next we create an example v1 Device for comparison purposes. It is here to show
+that you can mix v1 and v2 Devices in the same RunEngine:
+
+.. literalinclude:: ../examples/epics_demo.py
+    :language: python
+    :start-after: # Create v1
+    :end-before: # Create v2
+
+Finally we create the v2 Devices imported from the `epicsdemo` module:
+
+.. literalinclude:: ../examples/epics_demo.py
+    :language: python
+    :start-after: # Create v2
+
+The first thing to note is `with`. This uses a `DeviceCollector` as a context
+manager to collect up the top level `Device` instances created in the context,
+and run the following:
+
+- If ``set_name=True`` (the default), then call `Device.set_name` passing the
+  name of the variable within the context. For example, here we call
+  ``det.set_name("det")``
+- If ``connect=True`` (the default), then call `Device.connect` in parallel for
+  all top level Devices, waiting for up to ``timeout`` seconds. For example,
+  here we call ``asyncio.wait([det.connect(), samp.connect()])``
+- If ``sim=True`` is passed, then don't connect to PVs, but set Devices into
+  simulation mode
+- If ``prefix`` is passed, then use it as a prefix for all PV prefixes.
+
+The Devices we create in this example are a "sample stage" with a couple of
+"movers" called ``x`` and ``y`` and a "sensor" called ``det`` that gives a
+different reading depending on the position of the "movers".
+
+.. note::
+
+    There are very few Devices included in ophyd.v2, see the ophyd-epics-devices
+    and ophyd-tango-devices for some common ones associated with each control
+    system
+
+Run IPython
+-----------
+
+You can now run ipython with this startup file::
+
+    $ ipython -i epics_demo.py
+    IPython 8.5.0 -- An enhanced Interactive Python. Type '?' for help.
+
+    In [1]:
+
+.. ipython:: python
+    :suppress:
+
+    import sys
+    from pathlib import Path
+    sys.path.append(str(Path(".").absolute()/"docs/user_v2/examples"))
+    from epics_demo import *
+    # Turn off progressbar and table
+    RE.waiting_hook = None
+    bec.disable_table()
+
+This is like a regular python console with the contents of that file executed.
+IPython adds some extra features like tab completion and magics (shortcut
+commands). Ophyd adds some magics of its own that we will explore now.
+
+Try some magics
+---------------
+
+We can move the ``samp.x`` mover to 100mm using `bluesky.plan_stubs.mv`:
+
+.. ipython::
+
+    In [1]: mov samp.x 100
```

Comment:
Is this the way we want our users to interact with devices directly? Remember that this may be many users' first interactions with a motor device.

This is very different from python syntax to the point of being it's own language.

Reply:
Agreed.  Tou our users, it's a third syntax to learn (magics, ophyd methods, bluesky plans). And it is context specific to ipython/jupyter command lines, not scriptable.  Since I received this pushback from one instrument scientist, I've stopped showing the magics, in favor of the `samp.x.move(100)` syntax for command lines.

Reply:
To clarify what the magics actually do, they are a shorthand for running the plan under the run engine. So `mov samp.x 100` actually does `RE(bps.mov(samp.x, 100))`. This should be easily translatable to a plan:
```python
def my_plan():
    yield from bps.mov(samp.x, 100)
```
But I take your point that it is not scriptable and just another syntax to learn.

There's a reason for not implementing `motor.move()`: asyncio. There are good performance reasons for using it, but it means some changes. Things like `signal.read()` will now return a co-routine rather than the actual reading, this means you need to do `reading = await signal.read()` in the bluesky event loop in order to actually execute the code. This means there is no nice way to do the "dual duty" of `read()` and `set()` being both callable by the RE and on the commandline.

The options as I see it are:

### Magics
```ipython
from ophyd.v2 import magics
magics.register()
RE = ...
...
In [1]: mov samp.x 100  # calls RE(bps.mov(samp.x, 100))
```
### Helpers
```python
from ophyd.v2.helpers import mov
RE = ...
...
mov(samp.x, 100)  # calls RE(bps.mov(samp.x, 100))
```
### Extra Methods
```python
samp.x.move(100)  # calls asyncio.Task(get_bluesky_event_loop()).wait()
```
### Which one?
I'm not a big fan of the extra methods option, it muddies the Device level interface and it made things confusing for me when I started with Ophyd that you can `put` and `set` a Signal, and `move` and `set` a Motor. I'd prefer to keep the Ophyd Device interface pure Bluesky. It sounds like Magics will be contraversial, so maybe we're best off going with the helpers? It has the added bonus of directly translating to a plan in case those scriptable commands need to be embedded in a bigger plan.

@prjemian, @ZLLentz: do you think `mov(samp.x, 100)` be be acceptable instead of `samp.x.move(100)`?

Reply:
For users that interact with `ophyd` devices directly (which I understand is not recommended even in v1), the `samp.x.move(100)` syntax is important because it is easily discoverable for a non-expert. I know that I'm allowed to move a motor because it has a `move` method, and in `IPython` I can discover this by hitting `tab` after `samp.x.`

It's not a big deal for me if this isn't built into `ophyd` itself- I'll be adding the method to my motors for my users' preferences, and I accept that my users' preferences won't necessarily line up with the core community.

I do think having something like `mov(samp.x, 100)` is useful for keeping thing importable/pythonic regardless, but I don't think you should add it for my sake.

Reply:
I've backed out the tutorial to use just plan syntax while we have a wider discussion on this. I had a go at making a `motor.move()` method, and this is what I came up with:
```python
    def move(self, new_position: float, timeout: float = None):
        from bluesky.run_engine import call_in_bluesky_event_loop, in_bluesky_event_loop

        async def do_move():
            await self.set(new_position, timeout)

        assert not in_bluesky_event_loop(), "Will deadlock run engine if run in a plan, use mov(motor, pos) instead"
        call_in_bluesky_event_loop(do_move())
```
It will explicitly only work on the CLI as calling within a plan will block the asyncio event loop. The downside is that you don't get CTRL-C stopping the motor, as you're not actually running a plan. We don't have a handle on the RunEngine in this function, but if we did then we could run the plan using it. It feels a bit wrong coupling the Ophyd Devices back up to the run engine in this way though...

Reply:
Agree that talking to the RE from Ophyd is non-ideal. Makes sense that this functionality should exist and crash if someone tries to use it in a plan. Seems like the best compromise.